File: app/src/main/java/com/microsoft/msftbreweryworkshop/service/BeerService.java
package com.microsoft.msftbreweryworkshop.service;

import com.microsoft.msftbreweryworkshop.api.model.BeerDetail;
import com.microsoft.msftbreweryworkshop.api.model.BeerItem;
import com.microsoft.msftbreweryworkshop.api.model.DetailResponse;
import com.microsoft.msftbreweryworkshop.api.model.ListResponse;

interface BeerService {
    void getBeerList(ServiceListener<ListResponse<BeerItem>> listener);
    void getBeer(String id, ServiceListener<DetailResponse<BeerDetail>> listener);
}

File: app/src/main/java/com/microsoft/msftbreweryworkshop/service/BeerService.kt
package com.microsoft.msftbreweryworkshop.service

import com.microsoft.msftbreweryworkshop.model.Beer
import com.microsoft.msftbreweryworkshop.model.BeerListItem
import kotlinx.coroutines.Deferred

interface BeerService {
    fun getBeerList(): Deferred<List<BeerListItem>>
    fun getBeer(id: String): Deferred<Beer>
}